staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit.go

```diff
@@ -204,6 +213,10 @@ func (f *fancyResponseWriterDelegator) Flush() {
 func (f *fancyResponseWriterDelegator) Hijack() (net.Conn, *bufio.ReadWriter, error) {
 	// fake a response status before protocol switch happens
 	f.processCode(http.StatusSwitchingProtocols)
+
+	// TODO(caoshufeng): we can't call WriteHeader before Hijack(), because WriteHeader may be called later:
```

Comment:
Do we want to call WriteHeader (which is writing the code) or do we want only to add a header? The later is possible as long as `WriteHeader` was not called.